posts/2022-08-02-separate-stack-trace-22008.adoc

```diff
@@ -0,0 +1,466 @@
+----
++
+`server.xml` cofiguration
++
+[source,xml]
+----
+<featureManager>
+    <feature>jaxrs-2.0</feature>
+    <feature>ssl-1.0</feature>
+    <feature>openapi-3.1</feature>
+    <feature>concurrent-1.0</feature>
+    <feature>monitor-1.0</feature>
+    <feature>localConnector-1.0</feature>
+</featureManager>
+----
++ 
+This issue has since been reviewed and subsequently the affected features list have been updated once the UI is shared with the `mpOpenApi` features. This did previously work for openapi-3.1 but had been broken by link:https://github.com/OpenLiberty/open-liberty/pull/19535[#19535] in 22.0.0.2. This issue has now been resolved. 
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/21214[Server start fails when directory has spaces]
++ 
+When the server working directory is specified using environment variable SERVER_WORKING_DIR, and when a space is present in the name of the server working directory, an attempt to start the server using the Windows server script will hang, and the server will not start. 
++
+Steps to Reproduce are as follows:
++
+On Windows, set the server working directory using environment variable SERVER_WORKING_DIR with a directory name which contains a space. Then attempt to start the server using the server script `(server.bat)`.
++
+Note that this occurs on Windows only, Diagnostic information can be found below:
++
+OpenLiberty Versions: 21.0.0.6 and higher.
+Affected feature(s): All
+Java Version: All
+WINDOWS ONLY
++
+Additional Context for this issue can be found below:
++
+A work-around is available as the problem occurs because of an error in the server script "WLP_HOME/bin/server.bat".
++
+Quotes are missing near line 721:
++
+The line:
+`if x%SERVER_WORKING_DIR::=%==x%SERVER_WORKING_DIR% (`
++
+Should instead be:
+`if "%SERVER_WORKING_DIR::=%"=="%SERVER_WORKING_DIR%" (`
++
+A step of removing double quotes from SERVER_WORKING_DIR is missing near line 103.
++
+The line:
+`if defined WLP_OUTPUT_DIR set WLP_OUTPUT_DIR=!WLP_OUTPUT_DIR:"=!`
++
+Should be followed by the (completely new) line:
++
+`if defined SERVER_WORKING_DIR set SERVER_WORKING_DIR=!SERVER_WORKING_DIR:"=!`
++
+These two changes (updating one line; adding one completely new line) may be used to work-around the problem.
++
+The problem was introduced by the Pull Request:
++
+"JVM Working Directory Feature"
+#18931
+73b51bb
++
+Integrated into "gm-22.0.0.6-beta" on 22-Nov-2021.
++
+The Pull Request is for the Issue:
++
+"Add support for setting JVM working dir"
+#15128
++
+This is not a regression: The problem occurs relative to environment variable SERVER_WORKING_DIR, which was added by the issue 15128.
++
+This issue was identified as a work bug and worked on by the Open Liberty Kernal Team which has since been resolved. You can find out more information about this fix of adding quotes to server working directory test link:https://github.com/OpenLiberty/open-liberty/pull/21234[here].
```

Comment:
```suggestion
* link:https://github.com/OpenLiberty/open-liberty/issues/21214[Server start fails when directory has spaces]
+ 
On Windows operating system, when the server working directory is specified using environment variable SERVER_WORKING_DIR, and when a space is present in the name of the server working directory, an attempt to start the server using the server script will hang and the server will not start. 
+
This issue was due to missing quotes in the server script and has since been resolved. You can find out more information about this fix of adding quotes to server working directory test link:https://github.com/OpenLiberty/open-liberty/pull/21234[here].
```